src/prepare_problem.py

```diff
@@ -0,0 +1,123 @@
+"""This work was partially funded by the EuroHPC PL project,
+funded in the frame of Smart Growth Operational Programme, topic 4.2.
+
+Authors: Mateusz Hurbol, Justyna Zawalska
+"""
+
+import inspect
+import math
+from collections import defaultdict
+from dataclasses import dataclass
+from typing import Dict, List, Tuple
+
+import networkx as nx
+import pandas as pd
+import wfcommons
+from dataclasses_json import dataclass_json
+
+from src.utils.file_management import read_json_file
+
+
+@dataclass_json
+@dataclass
+class Task:
+    """
+    Class with the properties of a task.
+    """
+    id: str
+    name: str
+    machine: str
+    memory: float
+    runtime: float
+
+    @classmethod
+    def from_dict(cls, env):
+        """Allows to extract only selected fields that are needed to create the Task class."""
+
+        return cls(**{
+            k: v for k, v in env.items()
+            if k in inspect.signature(cls).parameters
+        })
+
+
+def get_machines_from_wfcommons_file(workflow_filename: str) -> Dict[str, dict]:
+    """Reads a WfCommons file that contains data about machines and
+    returns them in the form of a dict where the key is the machine
+    name and the values are machine details."""
+
+    machines_data = read_json_file(workflow_filename)
+    machines = {machine['nodeName']: machine for machine in machines_data['workflow']['machines']}
+    return machines
+
+
+def calc_cost(machine: dict, task: Task, runtime: float) -> float:
+    """Calculates the cost of performing the task on the selected machine."""
+
+    if task.memory is not None:
+        kib_free = machine['memory']
+        cost_power = (task.memory - kib_free) // (1024 * 1024)
+    else:
+        cost_power = 0
```

Comment:
```suggestion
    cost_power = (task.memory - machine['memory'])//(1024 * 1024) if task.memory else 0
```
Why is it divided by (1024 * 1024)? KiB is only 1024. Add comment on that please